### PR TITLE
CORE-18617: Add new parent key alias and created timestamp into key rotation status read

### PR DESCRIPTION
**Pull request by anien**
API PR: [1459](https://github.com/corda/corda-api/pull/1459)

### CODE REVIEW

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt [hunk 1]
```diff
@@ -199,7 +199,18 @@ class KeyRotationRestResourceImpl @Activate constructor(
             if (state.metadata[KeyRotationMetadataValues.STATUS] != KeyRotationStatus.DONE) rotationStatus =
                 KeyRotationStatus.IN_PROGRESS
         }
-        return KeyRotationStatusResponse(keyAlias, rotationStatus, lastUpdatedTimestamp, result)
+
+        // newParentKeyAlias and createdTimestamp are in all records, we just need to grab it from one
+        val deserializedValueOfOneRecord = requireNotNull(deserializer.deserialize(entries[entries.keys.first()]!!.value))
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: requireNotNull throws IllegalArgumentException you should really use it to check inputs to a method

Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: `entries.entries.first().value` will get you the first value in the map without having to do a null check or lookup by key, however for this reason I would consider renaming the `entries` val too

Comment (simon-johnson-r3) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: A follow up question, would you ever have more than one entry here?

**Review by simon-johnson-r3 — CHANGES_REQUESTED** [review]
Minor changes requested

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt (hunk 1 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: Records are grouped by tenantId, so yes, it's possible to have more than one entry.

Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: Changed to `checkNotNull`.

Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt]: I renamed `entries` to `records`.
PTAL.

**Review by simon-johnson-r3 — DISMISSED** [review]

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**